addon/index.js

```diff
@@ -186,6 +188,8 @@ function installExperiment(experiment) {
 }
 
 function uninstallSelf() {
+  // HACK: onUnload often doesn't seem to get called, here.
+  exports.onUnload('uninstall');
```

Comment:
And line 360 should be preventing the rest of that unload function from running twice, if it already got called on uninstall. That was the only thing that seemed to ensure the onunload happened reliably at least once on my machine